std/socket.d

```diff
@@ -1993,7 +2012,9 @@ static if (is(sockaddr_un))
 
         @property string path() @trusted const pure
         {
-            return to!string(cast(const(char)*)sun.sun_path.ptr);
+            return (cast(const(char)*) sun.sun_path.ptr)[0 ..
+                _nameLen - sockaddr_un.init.sun_path.offsetof -
+                (sun.sun_path.ptr[0] ? 1 : 0)].idup;
         }
```

Comment:
I see, thanks for the explanation. This should be added as a comment. I think it would be better to split the length calculation in multiple statements, so it's possible to explain each step.

A DDoc comment mentioning that filesystem paths are returned without the implicit null terminator but Linux abstract socket paths are returned as-is would be nice.